ledger/txtail.go

```diff
@@ -71,19 +79,44 @@ func (t *txTail) loadFromDisk(l ledgerForTracker) error {
 			return err
 		}
 
+		consensusParams := config.Consensus[blk.CurrentProtocol]
 		t.recent[old] = roundTxMembers{
-			txids:    make(map[transactions.Txid]basics.Round),
-			txleases: make(map[ledgercore.Txlease]basics.Round),
-			proto:    config.Consensus[blk.CurrentProtocol],
+			txleases: make(map[ledgercore.Txlease]basics.Round, len(payset)),
+			proto:    consensusParams,
 		}
+
 		for _, txad := range payset {
 			tx := txad.SignedTxn
-			t.recent[old].txids[tx.ID()] = tx.Txn.LastValid
-			t.recent[old].txleases[ledgercore.Txlease{Sender: tx.Txn.Sender, Lease: tx.Txn.Lease}] = tx.Txn.LastValid
-			t.putLV(tx.Txn.LastValid, tx.ID())
+			if consensusParams.SupportTransactionLeases && (tx.Txn.Lease != [32]byte{}) {
+				t.recent[old].txleases[ledgercore.Txlease{Sender: tx.Txn.Sender, Lease: tx.Txn.Lease}] = tx.Txn.LastValid
+			}
```

Comment:
can't this IF cause problems on early history replaying

Reply:
I don't believe it would cause issues. But I should definitely perform a full catchup to verify.
The case where we have an empty lease is never tested in `checkDup` below.